Source/Engine/Engine/InputEventState.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2012-2023 Wojciech Figat. All rights reserved.
+
+using System;
+
+namespace FlaxEngine
+{
+    /// <summary>
+    /// Virtual input action binding. Helps with listening for a selected input event.
+    /// </summary>
+    public class InputEventState
+    {
+        /// <summary>
+        /// The name of the action to use. See <see cref="Input.ActionMappings"/>.
+        /// </summary>
+        [Tooltip("The name of the action to use.")]
+        public string Name;
+
+        /// <summary>
+        /// Returns true if the event has been triggered during the current frame (e.g. user pressed a key). Use <see cref="Triggered"/> to catch events without active waiting.
+        /// </summary>
+        public bool Active => Input.GetAction(Name);
+
+        /// <summary>
+        /// Occurs when event is triggered (e.g. user pressed a key). Called before scripts update.
+        /// Has an additional parameter to read the input state.
+        /// </summary>
+        public event Action<InputActionState> Triggered;
```

Comment:
So why can't this be under `InputEvent` as `public event Action<InputActionState> StateChanged;` or something like that? I don't think adding a new action would be a breaking change.

Reply:
I can have a look at adding this, but normal InputEvents are handled the same way as regular InputActions and are only activated when in an exact state, the one that's specified in Input Settings. I initially tried to do this but I can have another look at adding it that way.